### PR TITLE
Fix many ruleset icons overflowing on triangles intro

### PR DESCRIPTION
**Pull request by Joehuu**
- Addresses https://github.com/ppy/osu/discussions/24077

| Before | After |
| --- | --- |
| osu Game Tests_xk7WbJdFde | osu Game Tests_2Em2dFA4LA |

### CODE REVIEW

**Comment by peppy** [comment]
I've adjusted this a bit to my own liking. Still dunno if it's required, having them go off the screen but still in one line looks more visually appeasing to me.

Will wait for a second opinion.

File: osu.Game/Screens/Menu/IntroTriangles.cs [hunk 1]
```diff
@@ -330,9 +332,10 @@ private void load(LargeTextureStore textures)
             private partial class RulesetFlow : FillFlowContainer
             {
                 [BackgroundDependencyLoader]
-                private void load(RulesetStore rulesets)
+                private void load(IRulesetStore rulesets)
                 {
-                    AutoSizeAxes = Axes.Both;
+                    Width = 800;
```
Comment (Joehuu) [inline osu.Game/Screens/Menu/IntroTriangles.cs:337]: Should use max size from grid container for portrait / UI scale scenarios.

Will do after we get that second opinion.

**Comment by bdach** [comment]
What hugely bothers me about this solution is the way the icons reflow during the animation. First they're



then they're



and finally



They shouldn't be doing that if you ask me. They should change scale, but not the way they flow. So if anything I kinda like the overflow more. And if we wanna do this, then it should be done in a way that the flowing doesn't happen again on the closeups.

**Comment by peppy** [comment]
> They shouldn't be doing that if you ask me. They should change scale, but not the way they flow. So if anything I kinda like the overflow more. And if we wanna do this, then it should be done in a way that the flowing doesn't change again on the closeups.

Agree with that. Was kind of my intention in fixing the width, but it doesn't work as well as I hoped and I really don't see this being something we should spend too much time on.

That said, if we're going to spend more time, it should probably be a custom flow to arrange them in a circle

```
   ***
  ****
   ***
```

etc.

**Comment by Joehuu** [comment]
Not going to spend time implementing the custom flow, so closing. May be more reasonable to fix once we have official custom rulesets.

**Pull request closed without merging**